### DISCUSSION TITLE
Caught Signal Number 11 SEGV ERROR

### DISCUSSION THREAD

**Discussion by usmanshahid90** [mdolab/dafoam#383] [Bug]
Hi,
I am trying to run my own mesh generated using ICEM. I have changed all the names of boundaries accordingly. but still I am getting this error. Complete error is:

Caught Signal Number 11 SEGV: Segmentation Violation Probably memory out of access...

Kindly guide,

Thanks

**Comment by uttamcadambi07** [comment]
This could be because of a lot of reasons. Please can you provide the details of your installation such as versions and so on? Also can you please elaborate on what stage you are receiving the error? I got something similar when I ran DAFoam for my case too and I suspect it is because I compiled petsc using 32 bit support and not 64 bit support so that could be the reason. But I am not sure.

**Reply by usmanshahid90** [reply]
Hey thanks for the reply. i got it cleared. What I've stucked in now is the mesh.
I am making 2D mesh in ICEM but when i bring it to openfoam, it doesn't have symmetry. instead, the front and back plane are combined as one name in openfoam.
In dafoam, we need to define symmetry plane. So how to get ICEM mesh in a way that we get front and back planed as seprate names.? kindly guide

**Reply by uttamcadambi07** [reply]
You can set the symmetry planes in dafoam based on point and normal. Assume that your domain goes from (bottom left back) <-10, -10, -0.05> to (top right front) <10, 10, 0.05>. The two planes that re front and back are defined as one - say sides for example. Set the boundary condition to be symmetry but in dafoam runScript.py make sure to set 
```
"symmetryPlanes": [[[0.0, 0.0, 0.05], [0.0, 0.0, 1.0]], [[0.0, 0.0, -0.05], [0.0, 0.0, -1.0]]],
```
This is what I am doing at least. Make sure to change the x, y and z coordinates based on your case.

**Reply by usmanshahid90** [reply]
thankyou so much. I will try it and let you know. Also do you have experience on Wing case?
My PC is without internet and i am using docker image. when i run wing case it asks for some files. Airfoil case is running well with same image but why not wing case?

**Reply by uttamcadambi07** [reply]
I am not sure of this sorry. Only @friedenhe  can answer that. I use DAFoam on my workstation and on my cluster so I do not have experience working with docker image.

But do let me know about the symmetry plane! I will also try it and we will see what happens.

**Reply by friedenhe** [reply]
> My PC is without internet and i am using docker image. when i run wing case it asks for some files. Airfoil case is running well with same image but why not wing case?

Start with the wing tutorial cases. It should work with Docker. Also, you need to check if you have enough memory for your case. Check the FAQ about the memory requirement.

**Reply by uttamcadambi07** [reply]
> Hey thanks for the reply. i got it cleared. What I've stucked in now is the mesh. I am making 2D mesh in ICEM but when i bring it to openfoam, it doesn't have symmetry. instead, the front and back plane are combined as one name in openfoam. In dafoam, we need to define symmetry plane. So how to get ICEM mesh in a way that we get front and back planed as seprate names.? kindly guide

How did you fix the SEGV error?

**Reply by usmanshahid90** [reply]
Hi, SEGV error was removed with eiditing registry. It was something, going into registry edit and then making some file with the name " ...pipeout".. and making the value of time equals to 180,000. 
I forgot exactly but i will share the link with you tommorow.
and by the way, thanks for the help. It worked. But the issue i faced was with the depth that openfoam takes in default. As because of that depth, mesh quality get degraded which fails the primal solution . so inorder to cater for it, we need to apply an OpenFoam command to scale down that extrusion which basically improves the mesh quality.
I'll remain in touch. You proved to be a great help, thankyou so much.

**Reply by uttamcadambi07** [reply]
Okay great.

By the way, you can manually fixed the depth problem.

Go into constant/polyMesh/points and do
```
vi
:%s/<z coordinate current value>/<z coordinate new value>/g
:wq
```
This way, you modify the dfault value (which i think runs from -0.24 to 0.24 so a depth of 0.48 which is very high). Set it to something like 0.05 so that your checkMesh does not report very high aspect ratio.

**Reply by usmanshahid90** [reply]
Alright. I'll definitely try it and will let you know. Also do u have idea about domain type? in default, dafoam uses circular domain but i am trying C type domain. Does it really makes amy difference? which one is better? what are your comments about it?

**Reply by uttamcadambi07** [reply]
Depends on your use. I use a C type domain that I meshed on fluent but a circular domain is also fine. I guess the dafoam uses automatic meshing tool to create the circular mesh for the airfoil. You can try to learn it and modify to suit your need but I dont have time so I use a C mesh. Also I am investigating stall so the downstream extent of the domain should be very long so as to not let the boundary conditions affect the solution so a C domain works well for me. If I used a circular domain I would unnecessarily increase the domain size in the upstream direction.

**Comment by usmanshahid90** [comment]
Sure.  Are u on slack?

**Reply by uttamcadambi07** [reply]
Yes

**Reply by usmanshahid90** [reply]
Have you joined the DAFOAM group?

**Reply by uttamcadambi07** [reply]
Yes I am on it

**Comment by usmanshahid90** [comment]
Hey, 
To remove SEGV error, what you need to do is go to:
HKEY_LOCAL_MACHINE\\SYSTEM\\CurrentControlSet\\Control\\
Over there, check for the registry "ServicesPipeTimeout". If there already exist a registry with this name then edit that registry and add a value of 180,000 in time. Else, create a registry with this name and then edit it. It solved this SEGV error in docker installation. I hope it would work for others as well. 
Thanks
